File: mint-runtime-permission/src/main/java/net/crystalapps/permission/runtime/library/callbacks/PermanentDeniedHandler.java
package net.crystalapps.permission.runtime.library.callbacks;


import androidx.annotation.NonNull;
import androidx.fragment.app.FragmentActivity;

import net.crystalapps.permission.runtime.library.annotations.Permit;

public interface PermanentDeniedHandler{
    void onHandle(@NonNull Permit permit, @NonNull FragmentActivity caller, @NonNull SettingOpener<FragmentActivity> settingOpener);
}

File: mint-runtime-permission/src/main/java/net/crystalapps/permission/runtime/library/callbacks/PermanentDeniedHandler.kt
package net.crystalapps.permission.runtime.library.callbacks


import androidx.fragment.app.FragmentActivity

import net.crystalapps.permission.runtime.library.annotations.Permit

interface PermanentDeniedHandler {
    fun onHandle(permit: Permit, caller: FragmentActivity, settingOpener: SettingOpener<FragmentActivity>)
}